Titanium.Web.Proxy/ProxyServer.cs

```diff
@@ -172,6 +172,10 @@ public CertificateEngine CertificateEngine
         /// </summary>
         public event Func<object, SessionEventArgs, Task> BeforeResponse;
 
+        public event Func<object, TunnelConnectEventArgs, Task> TunnelConnectRequest;
+
+        public event Func<object, TunnelConnectEventArgs, Task> TunnelConnectResponse;

```

Comment:
Yes, you are right, the response event is not called, yet. I'm still working on that.
I also plan to add some SSL information to the ConnectRequest/Response classes, line in another proxy, for example:
"A SSLv3-compatible ClientHello handshake was found. Extracted the parameters below.

Version: 3.3 (TLS/1.2)
Random: D3 FA 7E 31 4B EF 65 2E CA BA 3F AB 96 32 FE 42 53 DE 97 F9 11 B1 EC DB B9 60 B5 9F 52 63 C4 E4
"Time": 1996. 04. 25. 6:08:51
SessionID: 2C F7 B0 21 D9 83 81 A4 67 AA 4F 03 2C 4F 12 5A 33 52 F9 D6 3E 83 9F 1F 78 47 C0 8D 90 D4 AA 1E"

You can wait with the merge for that implementation, or temporary remove the TunnelConnectResponse event.